### PR TITLE
Support Gemini Experimental

### PR DESCRIPTION
**Pull request by DonggeLiu**
This PR supports `gemini-experimental`, which appears to work well in some cases.
Comparison experiment plan:
1. Compare `gemini-experimental` against our default `code-bison-32k` with the same temperature `0.4`.
2. Run `gemini-experimental` with its default temperature `1`.

### CODE REVIEW

**Comment by DonggeLiu** [comment]
@happy-qop, in case you are curious about the result : )

**Comment by DonggeLiu** [comment]
JOB: https://console.cloud.google.com/kubernetes/job/us-central1-c/llm-experiment/default/ofg-pr-234-dg
REPORT: https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-04-29-234-dg-comparison/index.html
BUCKET: https://console.cloud.google.com/storage/browser/oss-fuzz-gcb-experiment-run-logs/Result-reports/ofg-pr/2024-04-29-234-dg-comparison
BUCKET GS: `gs://oss-fuzz-gcb-experiment-run-logs/Result-reports/ofg-pr/2024-04-29-234-dg-comparison`

**Comment by DonggeLiu** [comment]
`code-bison-32k`, temperature `1`:
JOB: https://console.cloud.google.com/kubernetes/job/us-central1-c/llm-experiment/default/ofg-pr-234-dg-bison-temp-1
REPORT: https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-04-29-234-dg-bison-temp-1-comparison/index.html
BUCKET: https://console.cloud.google.com/storage/browser/oss-fuzz-gcb-experiment-run-logs/Result-reports/ofg-pr/2024-04-29-234-dg-bison-temp-1-comparison
BUCKET GS: `gs://oss-fuzz-gcb-experiment-run-logs/Result-reports/ofg-pr/2024-04-29-234-dg-bison-temp-1-comparison`

File: llm_toolkit/models.py [hunk 1]
```diff
@@ -350,6 +350,16 @@ class GeminiPro(GeminiModel):
   _vertex_ai_model = 'gemini-1.0-pro'
 
 
+class GeminiExperimental(VertexAIModel):
+  """Gemini Experimental, results maybe genius or delightfully unpredictable."""
```
Comment (oliverchang) [inline llm_toolkit/models.py:354]: Let's remove the "maybe genius or delightfully unpredictable." part :)  I know this is what is shown on Vertex AI, but no need to include it here in our codebase.

File: llm_toolkit/models.py [hunk 2]
```diff
@@ -350,6 +350,16 @@ class GeminiPro(GeminiModel):
...
+
+  _max_output_tokens = 8192
+  context_window = 32760
```
Comment (oliverchang) [inline llm_toolkit/models.py:357]: Not a big deal for our current workloads, but is this actually the same 1M context windows as Gemini 1.5?

File: llm_toolkit/models.py [hunk 3]
```diff
@@ -350,6 +350,16 @@ class GeminiPro(GeminiModel):
   _vertex_ai_model = 'gemini-1.0-pro'
 
 
+class GeminiExperimental(VertexAIModel):
```
Comment (oliverchang) [inline llm_toolkit/models.py:353]: I think this needs to inherit from `GeminiModel` instead.

File: llm_toolkit/models.py (hunk 2 above)
Comment (DonggeLiu) [reply llm_toolkit/models.py]: Done

File: llm_toolkit/models.py (hunk 1 above)
Comment (DonggeLiu) [reply llm_toolkit/models.py]: Done

**Comment by DonggeLiu** [comment]
The current quota for `gemini-experimental` is too low, will try again when the new quota application is approved.

**Comment by DonggeLiu** [comment]
/gcbrun request_pr_exp.py -n test-gemini-exp -m vertex_ai_gemini-experimental

**Pull request merged**